modules/management-console-agent/src/main/java/org/apache/ignite/agent/ws/WebSocketManager.java

```diff
@@ -259,136 +248,102 @@ private SslContextFactory createServerSslFactory(IgniteLogger log, ManagementCon
-    private void configureProxy(IgniteLogger log, HttpClient httpClient, URI srvUri) {
-        try {
-            boolean secure = "https".equalsIgnoreCase(srvUri.getScheme());
+    private void addAuthentication(ClientManager mgr) {
+        String user = System.getProperty("http.proxyUsername");
 
-            List<ProxyConfiguration.Proxy> proxies = ProxySelector.getDefault().select(srvUri).stream()
-                .filter(p -> !p.equals(NO_PROXY))
-                .map(p -> {
-                    InetSocketAddress inetAddr = (InetSocketAddress)p.address();
+        String pwd = System.getProperty("http.proxyPassword");
 
-                    Origin.Address addr = new Origin.Address(inetAddr.getHostName(), inetAddr.getPort());
+        if (!F.isEmpty(user) || !F.isEmpty(pwd)) {
+            Map<String, String> proxyHeaders = new HashMap<>();
 
-                    if (p.type() == SOCKS)
-                        return new Socks4Proxy(addr, secure);
+            proxyHeaders.put("Proxy-Authorization", "Basic " +
+                encodeToString((user + ':' + pwd).getBytes(Charset.forName("UTF-8"))));
 
-                    return new HttpProxy(addr, secure);
-                })
-                .collect(toList());
-
-            httpClient.getProxyConfiguration().getProxies().addAll(proxies);
-
-            addAuthentication(httpClient, proxies);
-        }
-        catch (Exception e) {
-            log.warning("Failed to configure proxy.", e);
+            mgr.getProperties().put(ClientProperties.PROXY_HEADERS, proxyHeaders);
         }
     }
 
     /**
-     * @param httpClient Http client.
-     * @param proxies Proxies.
+     * Trust all manager.
      */
-    private void addAuthentication(HttpClient httpClient, List<ProxyConfiguration.Proxy> proxies) {
-        proxies.forEach(p -> {
-            String user, pwd;
-
-            if (p instanceof HttpProxy) {
-                String scheme = p.getURI().getScheme();
+    private static class TrustAllManager implements X509TrustManager {
+        /** {@inheritDoc} */
+        @Override public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
 
-                user = System.getProperty(scheme + ".proxyUsername");
+        }
 
-                pwd = System.getProperty(scheme + ".proxyPassword");
-            }
-            else {
-                user = System.getProperty("java.net.socks.username");
+        /** {@inheritDoc} */
+        @Override public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
 
-                pwd = System.getProperty("java.net.socks.password");
-            }
+        }
 
-            httpClient.getAuthenticationStore().addAuthentication(
-                new BasicAuthentication(p.getURI(), ANY_REALM, user, pwd)
-            );
-        });
+        /** {@inheritDoc} */
+        @Override public X509Certificate[] getAcceptedIssuers() {
+            return null;
```

Comment:
Should return empty certificate array to prevent NPE. Look at DisabledX509TrustManager in ignite